store/sqlstore/post_store.go

```diff
@@ -648,47 +648,52 @@ func (s *SqlPostStore) getPostsAround(channelId string, postId string, numPosts
 		var posts []*model.Post
 		var parents []*model.Post
 		_, err1 := s.GetReplica().Select(&posts,
-			`(SELECT
+			`SELECT
 			    *
 			FROM
 			    Posts
 			WHERE
-				(CreateAt `+direction+` (SELECT CreateAt FROM Posts WHERE Id = :PostId)
+				CreateAt `+direction+` (SELECT CreateAt FROM Posts WHERE Id = :PostId)
 			        AND ChannelId = :ChannelId
-					AND DeleteAt = 0)
+					AND DeleteAt = 0
 			ORDER BY CreateAt `+sort+`
-			LIMIT :NumPosts
-			OFFSET :Offset)`,
-			map[string]interface{}{"ChannelId": channelId, "PostId": postId, "NumPosts": numPosts, "Offset": offset})
+			LIMIT :Limit
+			OFFSET :Offset`,
+			map[string]interface{}{"ChannelId": channelId, "PostId": postId, "Limit": limit, "Offset": offset})
 		_, err2 := s.GetReplica().Select(&parents,
-			`(SELECT
-			    *
+			`SELECT
```

Comment:
Might we be able to refactor `getParentsPosts` to accept a SQL Builder that defines the subquery defining from which root posts to select? (In general, rewriting all of this as a SQL Builder would help with clarity/composability.)

Reply:
That sounds like a good idea, but I'm not familiar with how the SQL builder stuff works or if it would be able to handle something nested and kind-of-messy like this